README.md

```diff
@@ -80,32 +79,19 @@ This function returns the initial state of store, which can be computed from the
 
 #### `effects: { [string]: Effect }`
 
-These functions can be called from application code (see `injectState`) and can do side-effects and/or mutate the state.
+These functions can be called from application code and can do side-effects and/or mutate the state.
 
 When called, an effect is provided with one or more arguments: a reference to other effects and any arguments passed to the effect from the application code.
 
-An effect can return:
-
-1. nothing (`undefined` or `null`)
-2. an object containing new properties to merge in the state
-3. a promise resolving to one of the above
-4. a function which receives the state and the props then returns one of the above
-
-```js
-{
-  effects: {
-    incrementCounter: (effects) => (state, props) => ({ counter: state.counter + 1 }),
-    onInputChange: (effects, event) => ({ counter: +event.target.value }),
-  }
-}
-```
-
-Effects can also access effects, state (read and write) and props (at the time the effect was called) via `this`, which is extremely
+Effects can access effects, state (read and write) and props (at the time the effect was called) via `this`, which is extremely
 handy for async effects:
 
+An effect returns always a promise resolving to `undefined` or `null`.
```

Comment:
Why is it important?

IMHO, you should add a example to show illustrate this.

Reply:
It is important in order for the users use always `await` before calling an `effect`

Reply:
> It is important in order for the users use always `await` before calling an `effect`

Exactly.

It does not matter that effects return promises, what matters is that they are asynchronous and must always be awaited. I think we could be a bit clearer and add an example.